src/models/quote.js

```diff
@@ -122,7 +125,26 @@ function * getAccountLedger (account) {
 
 function * getQuote (params, ledgers, config) {
   const query = yield makeQuoteQuery(params, config)
+  const type = query.source_amount ? 'source' : 'destination'
+  const amount = query.source_amount || query.destination_amount
   const quote = yield backend.getQuote(makeQuoteArgs(query))
+  try {
+    validate('Quote', quote)
```

Comment:
@lumberj added schema validation to the model. I'm re-throwing the error as a server error, since the validator returns an InvalidBodyError. I'm not sure that `ServerError` is the best error here, but I think it should be some kind of 500 error.